onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -0,0 +1,446 @@
+            onnx_attributes[key] = attr_proto.value
+
+    return onnx_inputs, onnx_attributes
+
+
+@beartype
+def _unwrap_tensor_to_torch_value(
+    value: ValidArgumentType
+    | dict[str, ValidArgumentType]
+    | list[ValidArgumentType]
+    | tuple[ValidArgumentType, ...]
+) -> torch.Value | str | int | float | dict[str, torch.Value | str | int | float] | list[
+    torch.Value | str | int | float
+] | tuple[torch.Value | str | int | float, ...]:
+    """Unwrap the TorchScriptTensor to torch.Value."""
+    if isinstance(value, dict):
+        return {
+            k: v.symbolic_value() if isinstance(v, TorchScriptTensor) else v
+            for k, v in value.items()
+        }
+    if isinstance(value, list):
+        return [v.symbolic_value() if isinstance(v, TorchScriptTensor) else v for v in value]
+    if isinstance(value, tuple):
+        return tuple(
+            v.symbolic_value() if isinstance(v, TorchScriptTensor) else v for v in value
+        )
+    if isinstance(value, TorchScriptTensor):
+        return value.symbolic_value()
+    # A normal python value
+    return value
+
+
+@beartype
+def _wrap_torch_value_to_tensor(
+    value: torch.Value
+    | dict[str, torch.Value | str | int | float]
+    | list[torch.Value | str | int | float]
+    | tuple[torch.Value | str | int | float, ...]
+) -> ValidArgumentType | dict[str, ValidArgumentType] | list[ValidArgumentType] | tuple[
+    ValidArgumentType, ...
+]:
+    """Wrap torch.Value to TorchScriptTensor."""
+    if isinstance(value, dict):
+        return {
+            k: TorchScriptTensor(v) if isinstance(v, torch.Value) else v
+            for k, v in value.items()
+        }
+    if isinstance(value, list):
+        return [TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value]
+    if isinstance(value, tuple):
+        return tuple(TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value)
+    if isinstance(value, torch.Value):
+        return TorchScriptTensor(value)
+    return value
+
+
+def _unwrap_tensors_to_torch_values(tensors):
+    # TODO(justinchuby): Do we really need this?
+    if isinstance(tensors, Sequence):
+        return [_unwrap_tensor_to_torch_value(output) for output in tensors]
+    return _unwrap_tensor_to_torch_value(tensors)
+
+
+class TorchScriptEvaluator(evaluator.Evaluator):
+    """An onnxscript Evaluator that captures the graph into torchscript."""
+
+    def __init__(self, graph: TorchScriptGraph):
+        self._graph: TorchScriptGraph = graph
+
+    @property
+    def graph(self) -> TorchScriptGraph:
+        return self._graph
+
+    @beartype
+    def eval_function(
+        self,
+        function: onnxscript.OnnxFunction,
+        args: Sequence[ValidArgumentType],
+        kwargs: Sequence[ValidArgumentType],
+    ):
+        # args/kwargs are TorchScriptTensor/python built-in based
+        inputs, attributes = _split_args_kwargs_to_input_attr(function, args, kwargs)
```

Comment:
Maybe it would be useful to do this in onnxscript *before* it calls the evaluator? Isn't this generic behavior we want shared by all evaluators?

Reply:
Good point. I was going to propose the same but thought it would be a longer discussion. Does the current split logic look good to you?